src/Adapter/ObjectModelComparator.php

```diff
@@ -0,0 +1,160 @@
+
+        $this->oldObject = $oldObject;
+        $this->newObject = $newObject;
+    }
+
+    /**
+     * Returns an array of two objectModels differences
+     *
+     * @return array
+     */
+    public function getDiff(): array
+    {
+        $differences = [];
+        $objectFields = array_merge($this->getObjectDefinedFields($this->oldObject), $this->getObjectNotDefinedFields());
+
+        foreach ($objectFields as $field => $definition) {
+            if (!property_exists($this->oldObject, $field)) {
+                continue;
+            }
+
+            $fieldType = $definition['type'] ?? null;
+
+            if (!empty($definition['lang']) && $definition['lang'] && is_array($this->newObject->$field)) {
+                foreach ($this->newObject->$field as $idLang => $newValue) {
+                    $oldValue = $this->oldObject->$field[$idLang] ?? null;
+
+                    if (!$this->fieldValueComparator($oldValue, $newValue, $fieldType)) {
+                        $differences[$field]['old'][$idLang] = $oldValue;
+                        $differences[$field]['new'][$idLang] = $newValue;
+                    }
+                }
+            } else {
+                if (!$this->fieldValueComparator($this->oldObject->$field, $this->newObject->$field, $fieldType)) {
+                    $differences[$field] = [
+                        'old' => $this->oldObject->$field,
+                        'new' => $this->newObject->$field,
+                    ];
+                }
+            }
+        }
+
+        return $differences;
+    }
+
+    public function hasChanges(): bool
+    {
+        return !empty($this->getDiff());
+    }
+
+    /**
+     * Returns old ObjectModel
+     *
+     * @return ObjectModel
+     */
+    public function getOldObject(): ObjectModel
+    {
+        return $this->oldObject;
+    }
+
+    /**
+     * Returns new ObjectModel
+     *
+     * @return ObjectModel
+     */
+    public function getNewObject(): ObjectModel
+    {
+        return $this->newObject;
+    }
+
+    /**
+     * Returns ObjectModel fields list
+     *
+     * @param ObjectModel $objectModel
+     *
+     * @return array
+     */
+    protected function getObjectDefinedFields(ObjectModel $objectModel): array
+    {
+        return $objectModel::$definition['fields'] ?? [];
+    }
+
+    /**
+     * Returns ObjectModel not defined fields list
+     *
+     * @return array
+     */
+    protected function getObjectNotDefinedFields(): array
+    {
+        return [
+            'id_shop_list' => ['type' => ObjectModel::TYPE_NOTHING],
+            'id_shop_default' => ['type' => ObjectModel::TYPE_INT],
+        ];
+    }
+
+    /**
+     * Compare field values
+     *
+     * @param $oldValue
+     * @param $newValue
+     * @param int|null $fieldType
+     *
+     * @return bool
+     */
+    private function fieldValueComparator($oldValue, $newValue, ?int $fieldType = null): bool
+    {
+        return $oldValue == $newValue;
```

Comment:
You should use `===` here instead and convert the values to DB (using DBAL/Type::convertToDatabase() method) instead.